src/components/TodoApp/TodoApp.tsx

```diff
@@ -0,0 +1,146 @@
+import React, { useEffect, useState } from 'react';
+import classNames from 'classnames';
+
+import { client } from '../../utils/fetchClient';
+import { URL, USER_ID } from '../../utils/Url';
+import { Todo } from '../../types/Todo';
+import { TodoList } from '../TodoList/TodoList';
+
+type Props = {
+  setTodos: React.Dispatch<React.SetStateAction<Todo[]>>
+  setAllTodos: React.Dispatch<React.SetStateAction<Todo[]>>
+  setErrorMessage: (a: string) => void;
+  todos: Todo[];
+  allTodos: Todo[];
+
+};
+
+export const TodoApp: React.FC<Props> = ({
+  todos,
+  allTodos,
+  setTodos,
+  setAllTodos,
+  setErrorMessage,
+}) => {
+  const [value, setValue] = useState<string>('');
+  const [isToggle, setIsToggle] = useState<boolean>(false);
+
+  useEffect(() => {
+    setTodos(allTodos);
+  }, [allTodos]);
+
+  const handleAddTodo = (event: React.ChangeEvent<HTMLInputElement>) => {
+    event.preventDefault();
+    const newValue: string = event.target.value;
+
+    setValue(newValue);
+  };
+
+  const addNewTodo = async () => {
+    if (!value.trim()) {
+      return;
+    }
+
+    try {
+      const newTodoData = {
+        title: value,
+        completed: false,
+        userId: USER_ID,
+        id: 0,
```

Comment:
The id of the new todo is set to 0. It would be better to generate a unique id for each new todo.